presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -607,16 +607,34 @@ void PrestoServer::initializeVeloxMemory() {
   const uint64_t memoryGb = systemConfig->systemMemoryGb();
   PRESTO_STARTUP_LOG(INFO) << "Starting with node memory " << memoryGb << "GB";
 
-  const int64_t memoryBytes = memoryGb << 30;
+  // Set up velox memory manager.
+  memory::MemoryManagerOptions options;
+  options.allocatorCapacity = memoryGb << 30;
   if (systemConfig->useMmapAllocator()) {
-    memory::MmapAllocator::Options options;
-    options.capacity = memoryBytes;
+    options.useMmapAllocator = true;
     options.useMmapArena = systemConfig->useMmapArena();
     options.mmapArenaCapacityRatio = systemConfig->mmapArenaCapacityRatio();
-    allocator_ = std::make_shared<memory::MmapAllocator>(options);
-  } else {
-    allocator_ = memory::MemoryAllocator::createDefaultInstance();
   }
+  options.checkUsageLeak = systemConfig->enableMemoryLeakCheck();
+  options.trackDefaultUsage =
+      systemConfig->enableSystemMemoryPoolUsageTracking();
+  if (!systemConfig->memoryArbitratorKind().empty()) {
+    options.arbitratorKind = systemConfig->memoryArbitratorKind();
+    const uint64_t queryMemoryGb = systemConfig->queryMemoryGb();
+    VELOX_USER_CHECK_LE(
+        queryMemoryGb,
+        memoryGb,
+        "Query memory capacity must not be larger than system memory capacity");
```

Comment:
Also add the actual values of queryMemoryGb and memoryGb in the error string.

Reply:
VELOX_USER_CHECK_LE will log the two values for comparison.